packages/tooltip/src/useTooltip.ts

```diff
@@ -47,13 +65,13 @@ export function useTooltip({ delayMilliseconds = 500, id, isVisible } = {}) {
     };
   }, []);
 
-  // Clean up stray timeouts if tooltip unmounts
+  // Clean up stray timeouts if tooltip un-mounts
   useEffect(() => {
     return () => {
       clearTimeout(openTooltipTimeout);
       clearTimeout(closeTooltipTimeout);
     };
-  }, [closeTooltipTimeout, openTooltipTimeout]);
+  }, [closeTooltipTimeout!, openTooltipTimeout!]);
```

Comment:
I added the `!` operator to tell TypeScript to assume these values are always defined. Otherwise, TypeScript would throw the following warning: 

> 🛑 Variable 'closeTooltipTimeout' is used before being assigned.ts(2454)

My original approach was to initialize `closeTooltipTimeout` and `openTooltipTimeout` to an empty timeout, and not use the `!` operator. I am not sure which is the idiomatic TypeScript way (or if there is a third alternative). I decided in the end to use the `!` operator. Open to suggestions and opinions!

```tsx
// initial approach
let openTooltipTimeout: NodeJS.Timeout = setTimeout(() => {}, 0);
let closeTooltipTimeout: NodeJS.Timeout = setTimeout(() => {}, 0);
```